engines/director/lingo/lingo.cpp

```diff
@@ -518,7 +506,8 @@ Common::String *Datum::toString() {
 		for (uint i = 0; i < u.farr->size(); i++) {
 			if (i > 0)
 				*s += ", ";
-			*s += *u.farr->operator[](i).toString();
+			Datum d = u.farr->operator[](i);
```

Comment:
Why? For readability?

Reply:
`u.farr->operator[](i)` returns a reference to the original array item, not just its value. Since `toString()` mutates the datum, `*u.farr->operator[](i).toString()` was mutating every array item into a string datum. Every time I tried to create an array of ints, it would suddenly become an array of strings when the stack was printed and `toString()` was called.

`Datum d = u.farr->operator[](i);` copies just the value into `d`. So then just the copy is cast to a string, and that copy is discarded after.

Reply:
To be honest I think mutating the datum in `toString()`, `toInt()`, etc. is an overall bad idea because it leads to issues like this one. Changing that would require a lot of fixes throughout the interpreter, though, which I haven't done that.

Reply:
Nice catch! And I have no other notes, great work, merging.